CONTRIBUTING.md

```diff
@@ -58,3 +58,47 @@ export const Page = ({ data, content }) => {
   )
 }
 ```
+
+## Internals
+
+At its core, Swingset's functionality revolves around a custom [Webpack loader](https://webpack.js.org/concepts/loaders). This loader enables Swingset to intercept imports for a number of different modules and inject custom metadata based on the detected component documentation. Swingset favors convention over configuration, and does its best to integrate into an existing Nextjs application without requiring significant structural changes or a ton of boilerplate code.
+
+### `(swingset)` route group
+
+The main "boilerplate" code is the `(swingset)` [route group](https://nextjs.org/docs/app/building-your-application/routing/route-groups). To isolate the Swingset styles and routes from the rest of your application, Swingset generates a route group with the following structure:
+
+```
+(swingset)/
+  - layout.tsx
+  - swingset/
+    - page.tsx
+    - [...path]/
+      - page.tsx
+    - docs/ # 👈 standalone documentation pages go here
+```
+
+Unless further customization is desired, The files generated here do not needed to be modified. Swingset will handle loading your documentation and the configured theme for rendering.
+
+### Entities
+
+Each detected piece of documentation content is represented within Swingset's core as an "entity." The base entity type contains information necessary to load and render a piece of documentation content. Currently, there are two types of entities: component entities and standalone documentation entities.
+
+### Module rules
+
+There are four distinct types of modules that, when requested, swingset will intercept:
+
+#### Content imports (`.mdx?`)
+
+Swingset will intercept `.mdx` and `.md` imports and compile the source into a render-able component. Internally, swingset dynamically imports each detected documentation file, and so those dynamically generated imports are then picked up by the this module rule.
+
+#### Component imports from content
+
+When importing component modules into documentation content, Swingset will attempt to parse out prop information and make it available for documenting.
```

Comment:
If i'm coming in with fresh eyes to the project I think I'd ask for more clarification here? 
Where / How would one access the parsed props?